tests/scalers_go/pulsar/pulsar_test.go

```diff
@@ -0,0 +1,255 @@
+  cooldownPeriod: 30 # Optional. Default: 300 seconds
+  maxReplicaCount: {{.MaxReplicaCount}}
+  minReplicaCount: {{.MinReplicaCount}}
+  triggers:
+    - type: pulsar
+      metadata:
+        msgBacklog: "{{.MsgBacklog}}"
+        activationTargetQueryValue: "5"
+        adminURL: http://{{.StatefulSetName}}.{{.TestNamespace}}:8080
+        topic:  persistent://public/default/keda
+        subscription: keda
+          `
+
+const publishJobTemplate = `
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{.ProducerJobName}}
+  namespace: {{.TestNamespace}}
+spec:
+  template:
+    spec:
+      containers:
+      - name: pulsar-client
+        image: ghcr.io/pulsar-sigs/pulsar-client:v0.3.1
+        imagePullPolicy: IfNotPresent
+        args: ["producer", "--broker","pulsar://{{.StatefulSetName}}.{{.TestNamespace}}:6650","--topic","persistent://public/default/keda","--message-num","{{.MessageCount}}"]
+      restartPolicy: Never
+  backoffLimit: 4
+`
+
+const serviceTemplate = `
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{.StatefulSetName}}
+  namespace: {{.TestNamespace}}
+spec:
+  type: ClusterIP
+  ports:
+  - name: http
+    port: 8080
+    targetPort: 8080
+    protocol: TCP
+  - name: pulsar
+    port: 6650
+    targetPort: 6650
+    protocol: TCP
+  selector:
+    app: pulsar
+`
+
+type templateValues map[string]string
+
+func getTemplateData() (templateData, templateValues) {
+	return templateData{
+			TestNamespace:          testNamespace,
+			StatefulSetName:        statefulSetName,
+			MessageCount:           messageCount,
+			ScaledObjectName:       scaledObjectName,
+			ConsumerDeploymentName: consumerDeploymentName,
+			ProducerJobName:        producerJobName,
+			MinReplicaCount:        minReplicaCount,
+			MaxReplicaCount:        maxReplicaCount,
+			MsgBacklog:             msgBacklog,
+		}, templateValues{
+			"statefulsetTemplate": statefulsetTemplate,
+			"serviceTemplate":     serviceTemplate,
+		}
+}
+
+func TestScaler(t *testing.T) {
+	// setup
+	t.Log("--- setting up ---")
+
+	// Create kubernetes resources
+	kc := GetKubernetesClient(t)
+	data, templates := getTemplateData()
+
+	CreateKubernetesResources(t, kc, testNamespace, data, templates)
+
+	assert.True(t, WaitForStatefulsetReplicaReadyCount(t, kc, statefulSetName, testNamespace, 1, 300, 1),
+		"replica count should be 1 after 5 minute")
+
+	KubectlApplyWithTemplate(t, data, "consumerTemplate", consumerTemplate)
+
+	// run consumer for create subcription
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, consumerDeploymentName, testNamespace, 1, 300, 1),
+		"replica count should be 1 after 5 minute")
+
+	KubectlApplyWithTemplate(t, data, "scaledObjectTemplate", scaledObjectTemplate)
+
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, consumerDeploymentName, testNamespace, 0, 60, 1),
+		"replica count should be 0 after a minute")
+
+	testActivation(t, kc, data)
+	KubectlDeleteWithTemplate(t, data, "publishJobTemplate", publishJobTemplate)
```

Comment:
I'd move this inside `testActivation` func because it's related with cleaning up the activation test